src/server/project.ts

```diff
@@ -760,6 +850,16 @@ namespace ts.server {
             return this.configFileName;
         }
 
+        getExternalFiles(): string[] {
+            const items: string[] = [];
+            for (const plugin of this.plugins) {
+                if (plugin.getExternalFiles) {
+                    items.push(...plugin.getExternalFiles(this));

```

Comment:
Yup. we need to wrapp this in a `try..catch` block to make sure the project is not in an invalid state.